gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt

```diff
@@ -8,11 +8,58 @@ package org.jetbrains.compose
 import org.gradle.api.Project
 import org.gradle.api.provider.Provider
 import org.jetbrains.compose.internal.ComposeCompilerArtifactProvider
+import org.jetbrains.compose.internal.KOTLIN_JVM_PLUGIN_ID
+import org.jetbrains.compose.internal.KOTLIN_MPP_PLUGIN_ID
+import org.jetbrains.compose.internal.Version
+import org.jetbrains.compose.internal.ideaIsInSyncProvider
 import org.jetbrains.compose.internal.mppExtOrNull
 import org.jetbrains.compose.internal.service.ConfigurationProblemReporterService
 import org.jetbrains.compose.internal.webExt
-import org.jetbrains.kotlin.gradle.plugin.*
+import org.jetbrains.kotlin.gradle.plugin.KotlinBasePlugin
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilerPluginSupportPlugin
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
+import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
+import org.jetbrains.kotlin.gradle.plugin.getKotlinPluginVersion
 import org.jetbrains.kotlin.gradle.targets.js.ir.KotlinJsIrTarget
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+internal fun Project.configureComposeCompilerPlugin() {
+    plugins.withId(KOTLIN_MPP_PLUGIN_ID) { plugin ->
+        configureComposeCompilerPlugin(plugin as KotlinBasePlugin)
+    }
+    plugins.withId(KOTLIN_JVM_PLUGIN_ID) { plugin ->
+        configureComposeCompilerPlugin(plugin as KotlinBasePlugin)
+    }
+}
+
+internal const val newCompilerIsAvailableVersion = "2.0.0-RC2"
+internal const val newComposeCompilerKotlinSupportPluginId = "org.jetbrains.kotlin.plugin.compose"
+internal const val newComposeCompilerError =
+    "Since Kotlin $newCompilerIsAvailableVersion to use Compose Multiplatform " +
+            "you must apply \"$newComposeCompilerKotlinSupportPluginId\" plugin."
```

Comment:
```suggestion
            "you must apply \"$newComposeCompilerKotlinSupportPluginId\" plugin. See the migration guide https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-compiler/compose-compiler.html#migrating-a-compose-multiplatform-project"
```

A preliminary link to the guide (draft is [here](https://helpserver.labs.jb.gg/help/kotlin-multiplatform-dev/compose-compiler/compose-compiler.html)). Let's merge it in the PR, and revert/change if needed before `beta03`.